pkg/model/application_live_state.go

```diff
@@ -144,3 +144,37 @@ func (s *ApplicationLiveStateSnapshot) determineLambdaAppHealthStatus() {
 	}
 	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
 }
+
+// DetermineApplicationHealthStatus updates the health status of the application based on the health status of its resources.
+func (s *ApplicationLiveStateSnapshot) DetermineApplicationHealthStatus() {
+	app := s.ApplicationLiveState
+	if app == nil {
+		return
+	}
```

Comment:
```suggestion
	if app == nil {
		s.HealthStatus = ApplicationLiveStateSnapshot_UNKNOWN
		return
	}
```
We should trickly assign the UNKNOWN stage here. The current logic looks like if this live state is nill, then the state will be kept "as is", which is unclearly UNKNOWN since UNKOWN is 0 in the enum. We should make it clear to avoid misunderstanding.

Reply:
FIxed at [f3bbedd](https://github.com/pipe-cd/pipecd/pull/5454/commits/f3bbedd0547098ab858e20c596cec00fcd990c98)